app/controllers/reviews_controller.rb

```diff
@@ -15,6 +15,27 @@ def update
     end
   end
 
+  def new
+    @shelter = Shelter.find(params[:shelter_id])
+  end
+
+  def create
+    @shelter = Shelter.find(params[:shelter_id])
+    review = @shelter.reviews.new(review_params)
+    if review.save
+      redirect_to "/shelters/#{params[:shelter_id]}"
+    else
+      flash[:notice] = "Review not created: Required information missing."
```

Comment:
We can consider `flash.now[:notice]` here so the error doesn't persist onto following pages.